src/core/ensure.rs

```diff
@@ -0,0 +1,14 @@
+use std::path::Path;
+use std::fs;
+use std::io;
+
+
+/// Ensure a directory exists and is empty
+pub fn ensure_dir_exists_fresh(dir: &Path) -> io::Result<()> {
```

Comment:
you could parametrise this over `<T>` that is `Into: Path` to get around my above comment.

Reply:
not that this really matters, just a nitpick

Reply:
Do you have some example syntax for this?

Reply:
yeah, don't do this actually. don't need to start the whole `Cow` usage for such small things.

Reply:
It's a thing usually only done in API boundaries with string and str: https://jwilm.io/blog/from-str-to-cow/

Reply:
it's overkill here anyway

Reply:
Yeah, this isn't an API boundary, this is just code reuse, but that's kinda nice.